content/_code-samples/trade-in-the-decentralized-exchange/js/trade-in-the-dex.js

```diff
@@ -0,0 +1,227 @@
+// Stand-alone code sample for "trade in the decentralized exchange" tutorial:
+// https://xrpl.org/trade-in-the-decentralized-exchange.html
+
+// Dependencies for Node.js.
+// In browsers, use <script> tags as in the example demo.html.
+if (typeof module !== "undefined") {
+  // Use var here because const/let are block-scoped to the if statement.
+  var xrpl = require('xrpl')
+  var BigNumber = require('bignumber.js')
+}
+
+// Connect ---------------------------------------------------------------------
+async function main() {
+  const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
+  console.log("Connecting to Testnet...")
+  await client.connect()
+
+  // Get credentials from the Testnet Faucet -----------------------------------
+  console.log("Requesting address from the Testnet faucet...")
+  const wallet = (await client.fundWallet()).wallet
+  console.log(`Got address ${wallet.address}.`)
+  // To use existing credentials, you can load them from a seed value, for
+  // example using an environment variable as follows:
+  // const wallet = xrpl.Wallet.fromSeed(process.env['MY_SEED'])
+
+  // Define the proposed trade. ------------------------------------------------
+  // Technically you don't need to specify the amounts (in the "value" field)
+  // to look up order books using book_offers, but for this tutorial we reuse
+  // these variables to construct the actual Offer later.
+  const we_want = {
+    currency: "TST",
+    issuer: "rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd",
+    value: "25"
+  }
+  const we_spend = {
+    currency: "XRP",
+    value: xrpl.xrpToDrops(25*10*1.15) // 25 TST * 10 XRP per TST * 15% fx cost
+  }
+  // "Quality" is defined as TakerPays ÷ TakerGets. The lower the "quality"
+  // number, the better the proposed exchange rate is for the taker.
+  // The quality is rounded to a number of significant digits based on the
+  // issuer's TickSize value (or the lesser of the two for token-token trades.)
+  const proposed_quality = BigNumber(we_spend.value) / BigNumber(we_want.value)
+
+  // Look up Offers. -----------------------------------------------------------
+  // To buy TST, look up Offers where "TakerGets" is TST:
+  const orderbook_resp = await client.request({
+    "command": "book_offers",
+    "taker": wallet.address,
+    "ledger_index": "current",
+    "taker_gets": we_want,
+    "taker_pays": we_spend
+  })
+  console.log(JSON.stringify(orderbook_resp.result, null, 2))
+
+  // Estimate whether a proposed Offer would execute immediately, and...
+  // If so, how much of it? (Partial execution is possible)
+  // If not, how much liquidity is above it? (How deep in the order book would
+  //    other Offers have to go before ours would get taken?)
+  // Note: these estimates can be thrown off by rounding if the token issuer
+  // uses a TickSize setting other than the default (15). In that case, you
+  // can increase the TakerGets amount of your final offer just a little bit to
```

Comment:
"just a little bit" is vague. Could we suggest a number or % instead?

Reply:
It's complicated: depends on the relative difference between the currencies involved, the TickRate setting, and how much the user is willing to pay above the _exact_ minimum. I'll just remove the words "just a little bit" since they're kind of redundant here anyway.